pkg/api/types.go

```diff
@@ -2379,6 +2380,12 @@ type DeleteOptions struct {
 	// Should the dependent objects be orphaned. If true/false, the "orphan"
 	// finalizer will be added to/removed from the object's finalizers list.
 	OrphanDependents *bool `json:"orphanDependents,omitempty"`
+
+	// Should the owner object keep existing in the key-value store until all dependents are removed.
+	// The option is cascading, i.e., the object’s dependents will be deleted with the same SynchronousGarbageCollection option.
+	// This option and OrphanDependents are exclusive.
+	// Defaults to false.
+	SynchronousGarbageCollection *bool
```

Comment:
I'd mildly prefer a field name that relates to the desired outcome - something like DeleteAfterDependents.  That's more consistent with the existing field and avoids introducing new words / concepts.